src/plugins/trimRows/trimRows.js

```diff
@@ -172,27 +172,37 @@ class TrimRows extends BasePlugin {
     const currentTrimConfig = this.getTrimmedRows();
     const isValidConfig = this.isValidConfig(rows);
     let destinationTrimConfig = currentTrimConfig;
+    const trimmingMapValues = this.trimmedRowsMap.getValues().slice();
+    const isAnyRowUntrimmed = rows.length > 0;
 
-    if (isValidConfig) {
-      destinationTrimConfig = currentTrimConfig.filter(trimmedRow => rows.includes(trimmedRow) === false);
+    if (isValidConfig && isAnyRowUntrimmed) {
+      // Preparing new values for trimming map.
+      arrayEach(rows, (physicalRow) => {
+        trimmingMapValues[physicalRow] = false;
+      });
+
+      // Preparing new trimming config.
+      destinationTrimConfig = arrayReduce(trimmingMapValues, (trimmedIndexes, isTrimmed, physicalIndex) => {
+        if (isTrimmed) {
+          trimmingMapValues.concat(physicalIndex);
```

Comment:
```suggestion
          trimmingMapValues.push(physicalIndex);
```
The same as [here](https://github.com/handsontable/handsontable/pull/7223/files#r481876518)